easybuild/easyblocks/i/impi.py

```diff
@@ -139,12 +139,16 @@ def post_install_step(self):
 
         impiver = LooseVersion(self.version)
         if impiver == LooseVersion('4.1.1.036') or impiver >= LooseVersion('5.0.1.035'):
+            if impiver >= LooseVersion('2018.0.128'):
+                script_paths = [os.path.join('intel64', 'bin')]
+            else:
+                script_paths = [os.path.join('intel64', 'bin'), os.path.join('mic', 'bin')]
             # fix broken env scripts after the move
-            for script in [os.path.join('intel64', 'bin', 'mpivars.csh'), os.path.join('mic', 'bin', 'mpivars.csh')]:
+            for script in [os.path.join(script_path,'mpivars.csh') for script_path in script_paths]:
                 for line in fileinput.input(os.path.join(self.installdir, script), inplace=1, backup='.orig.easybuild'):
                     line = re.sub(r"^setenv I_MPI_ROOT.*", "setenv I_MPI_ROOT %s" % self.installdir, line)
                     sys.stdout.write(line)
```

Comment:
@damianam While you're touching this, do you mind using `apply_regex_substitutions` instead like we already do in other easyblocks?

Reply:
Done. Hopefully without a type, I haven't tested it.